### PR TITLE
Sysdeps refactor

### PR DESCRIPTION
**Pull request by Geertiebear**
Should be a fix for #138

### CODE REVIEW

File: options/ansi/generic/environment.cpp [hunk 1]
```diff
@@ -106,7 +106,7 @@ char *getenv(const char *name) {
 	return const_cast<char *>(view.data() + s + 1);
 }
 
-int putenv(const char *string) {
+int __mlibc_putenv(char *string) {
```
Comment (avdgrinten) [inline options/ansi/generic/environment.cpp:109]: It would be nice to put this into the `mlibc::` namespace?

Comment (Geertiebear) [reply options/ansi/generic/environment.cpp:109]: Where would I put the declaration?

Comment (avdgrinten) [reply options/ansi/generic/environment.cpp:109]: Make some new ´environment.hpp` if nothing suitable exists?

**Comment by avdgrinten** [comment]
LGTM (but there are conflicts). I did not check that the sysdeps files are minimal but from a quick glance, they look fine. After this is merged, the logic behind the sysdeps headers should be documented in the README (and also explain why functions can appear in multiple sysdeps headers).

**Pull request merged**